logging/impl.go

```diff
@@ -131,8 +131,10 @@ func (imp *impl) AsZap() *zap.SugaredLogger {
 		}
 		ret = zap.Must(config.Build()).Sugar().Named(imp.name)
 	} else {
-		// `zaptest.NewLogger` always constructs a logger at the `zapcore.DebugLevel`.
-		ret = zaptest.NewLogger(testingObj, zaptest.WrapOptions(zap.AddCaller())).Sugar().Named(imp.name)
+		ret = zaptest.NewLogger(testingObj,
+			zaptest.WrapOptions(zap.AddCaller()),
+			zaptest.Level(imp.level.Get().AsZap()),
```

Comment:
Was this a bug where level wasn't getting passed down in the `-> *SugaredLogger` conversion?

Reply:
I wouldn't say it was a bug. My comment from January (correctly) stated that a `zaptest.NewLogger` always constructs at the debug log level. Note how that zaptest.NewLogger construction is entirely different than regular zap loggers. That create a config and call `Build` (see a couple lines above).

Because I had to learn a new API, I just didn't check hard enough to see if it was possible to change that behavior. With this test, I was motivated to look a little harder.

Reply:
Gotcha; that makes sense! Thanks.